NGitLab.Mock/Config/GitLabHelpers.cs

```diff
@@ -0,0 +1,1279 @@
+        {
+            return WithUser(config, username, user =>
+            {
+                user.Name = name;
+                user.Email = email;
+                user.AvatarUrl = avatarUrl;
+                user.IsAdmin = isAdmin;
+                if (asDefault)
+                {
+                    user.AsDefaultUser();
+                }
+
+                configure?.Invoke(user);
+            });
+        }
+
+        /// <summary>
+        /// Define a user as default for next descriptions
+        /// </summary>
+        /// <param name="user">User.</param>
+        public static GitLabUser AsDefaultUser(this GitLabUser user)
+        {
+            return Configure(user, _ =>
+            {
+                user.Parent.DefaultUser = user.Username;
+            });
+        }
+
+        /// <summary>
+        /// Define a user as default for next descriptions
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="username">Username.</param>
+        public static GitLabConfig SetDefaultUser(this GitLabConfig config, string username)
+        {
+            return Configure(config, _ =>
+            {
+                config.DefaultUser = username;
+            });
+        }
+
+        /// <summary>
+        /// Add an explicit group description in config
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="name">Name (required)</param>
+        /// <param name="configure">Configuration method</param>
+        public static GitLabConfig WithGroup(this GitLabConfig config, string name, Action<GitLabGroup> configure)
+        {
+            return Configure(config, _ =>
+            {
+                var group = new GitLabGroup
+                {
+                    Name = name ?? throw new ArgumentNullException(nameof(name)),
+                };
+
+                config.Groups.Add(group);
+                configure(group);
+            });
+        }
+
+        /// <summary>
+        /// Add an explicit group description in config
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="name">Name (required)</param>
+        /// <param name="id">Explicit ID (config increment)</param>
+        /// <param name="namespace">Parent namespace.</param>
+        /// <param name="description">Description.</param>
+        /// <param name="visibility">Visibility.</param>
+        /// <param name="defaultAsMaintainer">Define default user as maintainer.</param>
+        /// <param name="configure">Configuration method</param>
+        public static GitLabConfig WithGroup(this GitLabConfig config, string name, int id = default, string @namespace = null, string description = null, VisibilityLevel visibility = VisibilityLevel.Internal, bool defaultAsMaintainer = false, Action<GitLabGroup> configure = null)
```

Comment:
I think `defaultAsMaintainer` isn't clear enough. I had to read the code to understand what is does.

```suggestion
        public static GitLabConfig WithGroup(this GitLabConfig config, string name, int id = default, string @namespace = null, string description = null, VisibilityLevel visibility = VisibilityLevel.Internal, bool addDefaultUserAsMaintainer = false, Action<GitLabGroup> configure = null)
```

Reply:
```
VisibilityLevel visibility = VisibilityLevel.Internal
```

I think the default visibility should be set on the server (as GitLab), and reused here. Maybe it needs a new enum value `Default` (and a new enum)

Reply:
Shouldn't we have another extension to create subgroups like:

````
public static GitLabGroup WithGroup(this GitLabGroup @parentGroup, string name, ...)
````

Reply:
If you set `@namespace` variable, it will resolve parent or create it if not exists